### ISSUE TITLE
Difficult to move tabs between windows with collapsing vertical tabs enabled

### ISSUE THREAD

**Issue by FireStormOOO** [brave/brave-browser#40266]
### Description

It's taking a half-dozen attempts to move a tab (especially the last tab in a window) into an existing window.  The visual cues are there, but for whatever reason the tab just goes back into the old window (or a brand new window?) instead of merging.  It does eventually work but it's very touchy.  I'm really noticing this with the vertical tabs enabled and them collapsing automatically.

Tested on Debian 12 (Gnome, Wayland, AMD w/ Hardware Acceleration on).  It's touchy on Windows too but not quite like this.
Also tried on a Fedora install (Bazzite's Asus/Nvidia image, KDE); roughly same results.
There seems to be a performance aspect to this.  If the browser is snappy enough the issue is barely noticeable (e.g. nicely spec'd machine, freshly opened Brave with only a few tabs), but gets progressively more unusable as the browser bogs down even a little bit.

### Steps to reproduce

1. Enable vertical tabs; have them collapse to just the icons when not using the tab bar
2. Open several tabs in window A
3. Drag one of the tabs to a new window (B)
4. Attempt to drag the only tab in B back into A
5. Notice that most places you can drop the tab into A's tab bar don't work
6. It should eventually work after a few attempts

### Actual result

Most attempts to move window B's last tab back into A fail, with the tab either staying in the old window, or creating a new window instead.  Some attempts will work; I can't tell if the target is just much smaller than the visual area would indicate, or if there's another intermittent failure.

### Expected result

Upon seeing the visual cue for moving a tab, dropping the tab into the other window should work consistently.

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

Brave	1.67.123 Chromium: 126.0.6478.126 (Official Build) (64-bit) 
Revision	fc7619ef621fe4e6a0fa8b16c718f78ffc97a861
OS	Linux

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

No vertical tabs in chrome to compare with; we could maybe compare with Edge's implementation?
Obviously N/A on any per-tab features.
The worse the browser is performing overall, and the heavier the tab you're trying to move, the worse this gets.  For me it's ~5-6 tries to move a page with static web content.  I don't even try on something like YouTube.  The browser is otherwise working well, new Framework laptop with generous specs.

**Comment by parkeraddison** [comment]
It seems the issue here is that the **vertical tab bar collapses itself whenever a tab is moved via the mouse**. This happens mainly in two cases:

1. When moving a tab within a window, if you click in the middle of the tab and start to drag, then the tab bar collapses and the tab becomes a new window. Only once you drag the tab back into the bar will it stay expanded.

https://github.com/user-attachments/assets/91f74845-3941-4251-bc46-85026283fb04

2. If you move a tab into a window, the tab bar in the receiving window collapses (even when it should stay expanded) then expands again. This creates a confusing flickering effect.

https://github.com/user-attachments/assets/04cd215a-3565-4983-8003-d22057a66007

**There are two workarounds I've been using to deal with this issue:**
1. Remember to click and drag all the way at the left of the tab, over the favicon. Then the tab bar never expands. It's still fiddly with tabs near the start/end, because then often the tab creates a new window and the expand/collapse flickering occurs.

https://github.com/user-attachments/assets/dce596b7-0db9-4436-ad17-ef2d7a8885c6

 
2. **Tab movement functions as expected when the tab bar is manually expanded.** You can click the icon or set a shortcut to toggle the expanded state before moving tabs.

https://github.com/user-attachments/assets/0aa89d0c-b5a1-4410-9d4e-74ed77879dc8


### Brave version
Brave: 1.83.120 Chromium: 141.0.7390.122 (Official Build) (64-bit) 
Revision: 2647fda258705fce2a699977648e614d3e23e4ae
OS: Linux

**Comment by parkeraddison** [comment]
Quick note: The first workaround (drag a tab from its favicon) no longer works for fully hidden vertical tabs via _Settings > Tabs > Use vertical tabs > Hide completely when minimized_. Stick to being fiddly, or use a shortcut to expand the tab bar before reordering. Hope this helps someone.

**Comment by jquintanilla4** [comment]
yeah i've run into this a couple of times. I've just defaulted to using the right click menu when clicking on a tab. "Move tab to another window", it's not a great experience, but it works.
